pkg/resources/statefulsets/statefulset.go

```diff
@@ -109,9 +109,14 @@ func minioServerContainer(mi *miniov1beta1.MinIOInstance, serviceName, imagePath
 		"server",
 	}
 
-	// append all the MinIOInstance replica URLs
-	for i := 0; i < replicas; i++ {
-		args = append(args, fmt.Sprintf("%s://%s-"+strconv.Itoa(i)+".%s.%s.svc.cluster.local%s", scheme, mi.Name, serviceName, mi.Namespace, minioPath))
+	if replicas == 1 {
+		// to run in standalone mode we must pass the path
+		args = append(args, constants.MinIOVolumeMountPath)
+	} else {
```

Comment:
Shouldn't this be run as a `Deployment` instead of `StatefulSet` @jwilander

Reply:
Yes, good call @nitisht. I will update that.

Reply:
Actually after looking at the code again it would require non-trivial code changes to create a deployment for this one case rather than just using a single replica statefulset. I'm willing to make those changes but I'm not sure it's worth introducing significantly more code complexity to achieve that. Thoughts?

Reply:
Another thought, if we do continue to use a statefulset for standalone mode it might be easier to in the future add support for migrating from standalone to distributed.

Reply:
@jwilander standalone to distributed is a process of mc mirror - you cannot just change command line arguments to make it work.

Reply:
Right, but I could imagine having the operator create a job that could perform the `mc mirror` before increasing the replica count.

Reply:
>  I'm willing to make those changes but I'm not sure it's worth introducing significantly more code complexity to achieve that. Thoughts?